platform/security/core/security-core-impl/src/main/java/ddf/security/config/impl/ConfigurationSecurityLogger.java

```diff
@@ -0,0 +1,73 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p/>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p/>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package ddf.security.config.impl;
+
+import java.security.AccessController;
+
+import org.apache.shiro.subject.Subject;
+import org.apache.shiro.util.ThreadContext;
+import org.osgi.service.cm.Configuration;
+import org.osgi.service.cm.ConfigurationAdmin;
+import org.osgi.service.cm.ConfigurationEvent;
+import org.osgi.service.cm.SynchronousConfigurationListener;
+
+import ddf.security.common.audit.SecurityLogger;
+
+public class ConfigurationSecurityLogger implements SynchronousConfigurationListener {
+    private ConfigurationAdmin configurationAdmin;
+
+    @Override
+    public void configurationEvent(ConfigurationEvent event) {
+        try {
+            String type = getType(event);
+            Configuration configuration = configurationAdmin.getConfiguration(event.getPid());
+
+            Subject shiroSubject = ThreadContext.getSubject();
+            javax.security.auth.Subject javaSubject = javax.security.auth.Subject.getSubject(
+                    AccessController.getContext());
+            if (shiroSubject != null || javaSubject != null) {
+                SecurityLogger.audit("Configuration {}: {}",
+                        shiroSubject,
```

Comment:
Shouldn't this be either the `shiroSubject` or the `javaSubject`? Otherwise, you're checking that the `javaSubject` isn't null and then never using it and might have a null `shiroSubject` to log.